src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -59,13 +59,21 @@ class SetupPyRunner(InstallerBase):
 
   DIST_DIR = 'dist'
 
+  # NB: "snapshot" refers to a "snapshot release", not a Snapshot.
   @classmethod
-  def for_bdist_wheel(cls, source_dir, is_platform_specific, **kw):
-    cmd = ['bdist_wheel']
+  def for_snapshot_bdist_wheel(cls, source_dir, snapshot_fingerprint, is_platform_specific, **kw):
+    # NB: adds a '+' before the fingerprint to the build tag!
+    egg_info_snapshot_tag_args = ['egg_info', '--tag-build=+{}'.format(snapshot_fingerprint)]
+    bdist_whl_args = ['bdist_wheel']
     if is_platform_specific:
-      cmd.extend(['--plat-name', pep425tags.get_platform()])
-    cmd.extend(['--dist-dir', cls.DIST_DIR])
-    return cls(source_dir, cmd, **kw)
+      platform_args = ['--plat-name', pep425tags.get_platform()]
+    else:
+      platform_args = []
+    dist_dir_args = ['--dist-dir', cls.DIST_DIR]
+
+    setup_py_command = (
+      egg_info_snapshot_tag_args + bdist_whl_args + platform_args + dist_dir_args)
+    return cls(source_dir, setup_py_command, **kw)
```

Comment:
This all seems very specific to your use case. Any reason not to just move all this logic there and call the constructor directly?

Reply:
You're totally right. Will do that now.

Reply:
Moved to `_generate_snapshot_bdist_wheel_argv()` of `BuildLocalPythonDistributions` in 2aa1a5f!